packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts

```diff
@@ -0,0 +1,228 @@
+
+const useFetchThreadsQuery = (props: FetchBulkThreadsProps | FetchActiveThreadsProps) => {
+  const { chainId } = props
+
+  // better to use this in case someone updates this props, we wont reflect those changes
+  const [queryType] = useState(props.queryType)
+
+  const chosenQueryType = queryTypeToRQMap[queryType]({
+    queryKey: (() => {
+      if (isFetchBulkThreadsProps(props)) {
+        return [
+          ApiEndpoints.FETCH_THREADS,
+          chainId,
+          queryType,
+          props.topicId,
+          props.stage,
+          props.includePinnedThreads,
+          props.toDate,
+          props.fromDate,
+          props.limit,
+          props.orderBy
+        ]
+      }
+      if (isFetchActiveThreadsProps(props)) {
+        return [
+          ApiEndpoints.FETCH_THREADS,
+          chainId,
+          queryType,
+          props.topicsPerThread
+        ]
+      }
+    })(),
+    queryFn: (() => {
+      if (isFetchBulkThreadsProps(props)) {
+        return async ({ pageParam = 1 }) => {
+          const res = await axios.get(
+            `${app.serverUrl()}${ApiEndpoints.FETCH_THREADS}`,
+            {
+              params: {
+                bulk: true,
+                page: pageParam,
+                limit: props.limit,
+                chain: chainId,
+                ...(props.topicId && { topic_id: props.topicId }),
+                ...(props.stage && { stage: props.stage }),
+                ...(props.includePinnedThreads && { includePinnedThreads: props.includePinnedThreads || true }),
+                ...(props.fromDate && { from_date: props.fromDate }),
+                to_date: props.toDate,
+                orderBy:
+                  featuredFilterQueryMap[props.orderBy] ||
+                  featuredFilterQueryMap.newest,
+              },
+            }
+          );
+
+          // transform the response
+          const transformedData = {
+            ...res.data.result,
+            threads: res.data.result.threads.map((c) => new Thread(c))
+          };
+
+          return { data: transformedData, pageParam: transformedData.threads.length > 0 ? pageParam + 1 : undefined }
+        }
+      }
+      if (isFetchActiveThreadsProps(props)) {
+        return async () => {
+          const response = await axios.get(
+            `${app.serverUrl()}${ApiEndpoints.FETCH_THREADS}`,
+            {
+              params: {
+                active: true,
+                chain: chainId,
+                threads_per_topic: props.topicsPerThread || 3,
+              },
+            }
+          );
+
+          // transform response
+          return response.data.result.map((c) => new Thread(c))
+        }
+      }
+    })(),
+    ...(() => {
+      if (isFetchBulkThreadsProps(props)) {
+        return {
+          getNextPageParam: (lastPage, pages) => lastPage.pageParam,
+        }
+      }
+    })(),
```

Comment:
too much going on here, a lot of parameters, brackets, desctructurings etc. 

my suggestion:
```
if (isFetchBulkThreadsProps){
  fetchBulkThreads()
} 

if (isFetchActiveThreadsProps) {
  fetchActiveThreads()
}

```

we will gain a lot of readability by doing this

Reply:
Done